libcnb-data/src/build_plan.rs

```diff
@@ -42,6 +42,12 @@ impl BuildPlanBuilder {
         self
     }
 
+    /// Add Require to the `BuildPlan`
+    pub fn requires_with_struct(mut self, require: Require) -> Self {
```

Comment:
I find the `requires_with_struct` name a little clunky. Maybe we can change the `requires` signature to this instead:

```rust
pub fn requires<R: Into<Require>>(mut self, require: R) -> Self {
    self.current_requires.push(require.into());
    self
}
```

and then provide `From<String> for Require` and `From<&str> for Require`. Since `From` is reflexive (`From<T> for T` is always available), this would allow the user to use either the full struct or just a string in `requires` without an additional function that is hard to name.

```rust
impl<S: Into<String>> From<S> for Require {
    fn from(s: S) -> Self {
        Require {
            name: s.into(),
            metadata: Default::default(),
        }
    }
}

impl<S: Into<String>> From<S> for Provide {
    fn from(s: S) -> Self {
        Provide { name: s.into() }
    }
}
```

WDYT?